cibuildwheel/pyodide.py

```diff
@@ -66,6 +67,47 @@ def install_emscripten(tmp: Path, version: str) -> Path:
     return emcc_path
 
 
+def get_xbuildenv_versions(env: dict[str, str]) -> list[str]:
+    """Searches for the compatible xbuildenvs for the current pyodide-build version"""
+    with FileLock(CIBW_CACHE_PATH / "xbuildenv.lock"):
+        xbuildenvs = call(
+            "pyodide",
+            "xbuildenv",
+            "search",
+            "--json",
+            "--all",
+            env=env,
+            cwd=CIBW_CACHE_PATH,
+            capture_stdout=True,
+        ).strip()
+        xbuildenvs_dict = json.loads(xbuildenvs)
+        compatible_xbuildenv_versions = [
+            _["version"] for _ in xbuildenvs_dict["environments"] if _["compatible"]
+        ]
+
+        return compatible_xbuildenv_versions
```

Comment:
I'm just coming back to this, but can we use more of the output of xbuildenv here? It also specs the enscripten version. That is currently hard-coded in `build-platforms.toml`. Also currently hard-coded in build-platforms.toml are pyodide-build version - which is now handled by CIBW_DEPENDENCY_VERSIONS. So I think we could strip down the `build-platforms.toml` entry to just Python version, node version and default pyodide version. 

That does make me wonder if there's a strong link between Python version and node version... or would node version also need to be customisable?

Reply:
> if there's a strong link between Python version and node version... or would node version also need to be customisable?

I don't think there will be any reason to prefer old versions of node. There may be reasons to prefer newer ones.